File: 20160206_HomeWorkCpp/20160206_HomeWorkCpp/HomeWorkHelper.cpp
#include "stdafx.h"
#include "HomeWorkHelper.h"

HomeWorkHelper::HomeWorkHelper() {}

HomeWorkHelper::~HomeWorkHelper() {}

void HomeWorkHelper::PrintHelloWorld(int num)
{
	for (int i = 0; i < num; i++)
	{
		cout << "Hello, world!" << endl;
	}
}

void HomeWorkHelper::Average(int num1, int num2)
{
	printf("The average for '%d' and '%d' is '%d'.", num1, num2, (num1 + num2) / 2);
}

void HomeWorkHelper::ReturnNumberOfDigits(int num)
{
	int digitsCounter = 0;
	int tempNumber = num;
	bool isEnded = false;

	while (!isEnded)
	{
		if ((tempNumber / 10) >= 0 && (tempNumber % 10) > 0)
		{
			tempNumber /= 10;
			digitsCounter++;
		}
		else
		{
			isEnded = true;
		}
	}

	printf("The number of digits for number '%d' is '%d'.", num, digitsCounter);
}

void HomeWorkHelper::WriteNumberRange(int num)
{
	for (int i = 0; i < num; i++)
	{
		cout << i << " ";
	}
}

bool HomeWorkHelper::IsNumberNegative(int num)
{
	bool isNegativeNumber = false;

	if (num == 0)
	{
		printf("The number '%d' is zero.", num);
		return false;
	}

	if (num < 0)
	{
		isNegativeNumber = true;
	}

	printf("The number '%d' negative? '%s'.", num, isNegativeNumber ? "true" : "false");	
	return isNegativeNumber;
}

void HomeWorkHelper::PlaySound(int num)
{
	const char beep = '\a';
	for (int i = 0; i < num; i++)
	{
		cout << beep;
	}
}

void HomeWorkHelper::FlipAbsForNumber(int num)
{
	if (num <= 0)
	{
		cout << abs(num) << endl;
	}

	if (num > 0)
	{
		cout << (num *= -1) << endl;
	}
}

int HomeWorkHelper::ReturnAbsForNumber(int num)
{
	int absNum = abs(num);
	printf("Abs for number '%d' is '%d'.", num, absNum);
	return absNum;
}

bool HomeWorkHelper::IsNumberCanBeDividedByNumber(int num1, int num2)
{
	bool result = false;

	if (num2 != 0)
	{
		result = num1 / num2 > 0;
		printf("The number '%d' can be divided by number '%d'? '%s'.", num1, num2, result ? "true" : "false");
	}
	else
	{
		throw exception("Division by zero error!\nCheck your data!");
	}

	return result;
}

int HomeWorkHelper::ReturnRandomNumberForRange()
{
	srand((unsigned)time(NULL));
	int result = rand() % 13 + 1;
	printf("Random number for range from '%d' to '%d' is '%d'.", 1, 13, result);

	return result;
}

void HomeWorkHelper::ShowDivisionResult(int num1, int num2)
{
	if (num2 != 0)
	{
		printf("Division result for numbers: '%d' and '%d' is '%f'.", num1, num2, ((float)num1 / num2));
	}
	else
	{
		throw exception("Divide by zero error!\nCheck your data!");
	}
}

int HomeWorkHelper::ReturnModuloForNumbers(int num1, int num2)
{
	int result = 0;

	if (num1 == num2 && num1 == 0)
	{
		printf("Modulo for numbers: '%d' and '%d' is '%d'.", num1, num2, 0);
		return 0;
	}

	if (num1 > num2)
	{
		if (num2 != 0)
		{
			result = num1 % num2;
		}
		else
		{
			throw exception("Divide by zero error!\nCheck your data!");
		}
	}

	if (num2 > num1)
	{
		if (num1 != 0)
		{
			result = num2 % num1;
		}
		else
		{
			throw exception("Divide by zero error!\nCheck your data!");
		}
	}

	printf("Modulo for numbers: '%d' and '%d' is '%d'.", num1, num2, result);
	return result;
}

File: old/20160206_HomeWork/20160206_HomeWork/HomeWorkHelper.cs
using System;

namespace _20160206_HomeWork
{
    /// <summary>
    /// This is homework helper.
    /// Most of numbers are integers for simplicity.
    /// </summary>
    class HomeWorkHelper
    {
        public void PrintHelloWorld(int num)
        {
            for (int i = 0; i < num; i++)
            {
                Console.WriteLine("Hello, world!");
            }
        }

        public void Average(int num1, int num2)
        {
            Console.WriteLine("The average for {0} and {1} is {2}.", num1, num2, (num1 + num2) / 2);
        }

        public void ReturnNumberOfDigits(int num)
        {
            int digitsCounter = 0;
            int tempNumber = num;
            bool isEnded = false;

            while (!isEnded)
            {
                if ((tempNumber / 10) >= 0 && (tempNumber % 10) > 0)
                {
                    tempNumber /= 10;
                    digitsCounter++;
                }
                else
                {
                    isEnded = true;
                }
            }

            Console.WriteLine("The number of digits for number {0} is {1}", num, digitsCounter);
        }

        public void WriteNumberRange(int num)
        {
            for (int i = 0; i < num; i++)
            {
                Console.Write(i + " ");
            }
        }

        public bool IsNumberNegative(int num)
        {
            bool isNegativeNumber = false;

            if (num == 0)
            {
                Console.WriteLine("The number '{0}' is zero.", num);
                return false;
            }

            if (num < 0)
            {
                isNegativeNumber = true;
            }

            Console.WriteLine("The number '{0}' negative? {1}", num, isNegativeNumber);

            return isNegativeNumber;
        }

        public void PlaySound(int num)
        {
            for (int i = 0; i < num; i++)
            {
                Console.Beep();
            }
        }

        public void FlipAbsForNumber(int num)
        {
            if (num <= 0)
            {
                Console.WriteLine(Math.Abs(num));
            }

            if (num > 0)
            {
                Console.WriteLine(num *= -1);
            }
        }

        public int ReturnAbsForNumber(int num)
        {
            int absNum = Math.Abs(num);
            Console.WriteLine("Abs for number '{0}' is '{1}'.", num, absNum);
            return absNum;
        }

        public bool IsNumberCanBeDividedByNumber(int num1, int num2)
        {
            bool result = false;

            if (num2 != 0)
            {
                result = num1 / num2 > 0;
                Console.WriteLine("The number '{0}' can be divided by number '{1}'? {2}.", num1, num2, result);
            }
            else
            {
                throw new Exception("Division by zero error!\nCheck your data!");
            }

            return result;
        }

        public int ReturnRandomNumberForRange()
        {
            Random random = new Random();
            int result = random.Next(1, 13);
            Console.WriteLine("Random number for range from '{0}' to '{1}' is '{2}'.", 1, 13, result);
            return result;
        }

        public void ShowDivisionResult(int num1, int num2)
        {
            if (num2 != 0)
            {
                Console.WriteLine("Division result for numbers: '{0}' and '{1}' is {2}.", num1, num2, ((float)num1 / num2));
            }
            else
            {
                throw new Exception("Divide by zero error!\nCheck your data!");
            }
        }

        public int ReturnModuloForNumbers(int num1, int num2)
        {
            int result = 0;

            if (num1 == num2 && num1 == 0)
            {
                Console.WriteLine("Modulo for numbers: '{0}' and '{1}' is '{2}'.", num1, num2, 0);
                return 0;
            }

            if (num1 > num2)
            {
                if (num2 != 0)
                {
                    result = num1 % num2;
                }
                else
                {
                    throw new Exception("Divide by zero error!\nCheck your data!");
                }
            }

            if (num2 > num1)
            {
                if (num1 != 0)
                {
                    result = num2 % num1;
                }
                else
                {
                    throw new Exception("Divide by zero error!\nCheck your data!");
                }
            }

            Console.WriteLine("Modulo for numbers: '{0}' and '{1}' is '{2}'.", num1, num2, result);
            return result;
        }
    }
}
